app/javascript/helpers/forms/SignupFormHelpers.jsx

```diff
@@ -5,8 +5,17 @@ const validationSchema = yup.object({
   // TODO: amir - Revisit validations.
   name: yup.string().required('Please enter your full name.'),
   email: yup.string().required('Please enter your email.').email('Entered value does not match email format.'),
-  password: yup.string().required('Please enter your password.').min(8, 'Password must have at least 8 characters.'),
-  password_confirmation: yup.string().oneOf([yup.ref('password')], 'Your passwords do not match.'),
+  password: yup.string()
+    .matches(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*(\W|_)).{8,}$/,
+      'Password must have at least:',
+    )
+    .min(8, '- Eight charachters.')
+    .test('oneLower', '- One lowercase letter.', (pwd) => pwd.match(/^.*[a-z].*$/))
+    .test('oneUpper', '- One uppercase letter.', (pwd) => pwd.match(/^.*[A-Z].*$/))
+    .test('oneDigit', '- One digit.', (pwd) => pwd.match(/^.*\d.*$/))
+    .test('oneSpecial', '- One none alphanumeric charachter.', (pwd) => pwd.match(/^.*(\W|_).*$/)),
```

Comment:
I would say "one symbol" not "one alphanumeric character"

Reply:
I would also remove the `.` at the end of each line

Reply:
The regex pattern is `\W|_` that accepts any no alphanumeric charachter so 'é' will be treated as a symbol.

Reply:
Should we change the pattern (that's what we avoided on v2) or we keep it as it but we indicate a symbol which is a subset of the accepted combinations?

Reply:
I'd prefer we specify symbol rather than alphanumeric character